src/Mailbox.php

```diff
@@ -247,4 +240,54 @@ public function getThread(): array
 
         return false !== $tree ? $tree : [];
     }
+
+    /**
+     * Bulk move messages.
+     *
+     * @param array|MessageIteratorInterface|string $numbers Message numbers
+     * @param MailboxInterface                      $mailbox Destination Mailbox to move the messages to
+     *
+     * @throws \Ddeboer\Imap\Exception\MessageMoveException
+     */
+    public function move($numbers, MailboxInterface $mailbox)
+    {
+        if (!\imap_mail_move($this->resource->getStream(), $this->prepareMessageIds($numbers), $mailbox->getEncodedName(), \CP_UID)) {
+            throw new MessageMoveException(\sprintf('Messages cannot be moved to "%s"', $mailbox->getName()));
+        }
+    }
+
+    /**
+     * Bulk copy messages.
+     *
+     * @param array|MessageIteratorInterface|string $numbers Message numbers
+     * @param MailboxInterface                      $mailbox Destination Mailbox to copy the messages to
+     *
+     * @throws \Ddeboer\Imap\Exception\MessageMoveException
+     */
+    public function copy($numbers, MailboxInterface $mailbox)
+    {
+        if (!\imap_mail_copy($this->resource->getStream(), $this->prepareMessageIds($numbers), $mailbox->getEncodedName(), \CP_UID)) {
+            throw new MessageMoveException(\sprintf('Messages cannot be copied to "%s"', $mailbox->getName()));
+        }
+    }
+
+    /**
+     * Prepare message ids for the use with bulk functions.
+     *
+     * @param array|MessageIteratorInterface|string $messageIds Message numbers
```

Comment:
`getArrayCopy` isn't a method of `MessageIteratorInterface` but only of `MessageIterator`: please change all the doc hints to the latter class

Reply:
right. fixed

Reply:
Or not. This is now breaking the build since the static analyser complains. I could fix this by typehinting the returned variable in the test - but is that such a good idea?

```
diff --git a/tests/MailboxTest.php b/tests/MailboxTest.php
index 89cd7c5..84a58da 100644
--- a/tests/MailboxTest.php
+++ b/tests/MailboxTest.php
@@ -10,6 +10,7 @@ use Ddeboer\Imapxception\MessageDoesNotExistException;
 use Ddeboer\Imapxception\MessageMoveException;
 use Ddeboer\Imapxception\ReopenMailboxException;
 use Ddeboer\Imap\MailboxInterface;
+use Ddeboer\Imap\MessageIterator;
 
 /**
  * @covers \Ddeboer\Imapxception\AbstractException
@@ -234,6 +235,7 @@ final class MailboxTest extends AbstractTest
         $this->assertSame(0, $this->mailbox->count());
 
         // move back by iterator
+        /** @var MessageIterator $messages */
         $messages = $anotherMailbox->getMessages();
         $anotherMailbox->move($messages, $this->mailbox);
         $this->getConnection()->expunge();
```

Reply:
> but is that such a good idea?

The bad thing is `getArrayCopy` isn't present in any SPL interface, and this complicates everything a lot.
In a future release we could add it to `MessageIteratorInterface` but at the time being your solution is enough.